src/nupic/types/Sdr.cpp

```diff
@@ -368,6 +384,44 @@ namespace sdr {
         SDR::setDenseInplace();
     }
 
+
+    void SparseDistributedRepresentation::set_union(
+            const SDR &input1, const SDR &input2) {
+        set_union( { &input1, &input2 } );
+    }
+
+    void SparseDistributedRepresentation::set_union(vector<const SDR*> inputs) {
+        NTA_CHECK( inputs.size() >= 2u );
+        bool inplace = false;
+        for( size_t i = 0; i < inputs.size(); i++ ) {
+            NTA_CHECK( inputs[i] != nullptr );
+            NTA_CHECK( inputs[i]->dimensions == dimensions );
+            // Check for modifying this SDR inplace.
+            if( inputs[i] == this ) {
+                inplace = true;
+                inputs[i--] = inputs.back();
+                inputs.pop_back();
+            }
+        }
+        if( inplace ) {
+            getDense(); // Make sure that the dense data is valid.
+        }
+        if( not inplace ) {
+            // Copy one of the SDRs over to the output SDR.
+            const auto &denseIn = inputs.back()->getDense();
+            dense_.assign( denseIn.begin(), denseIn.end() );
+            inputs.pop_back();
+            // inplace = true; // Now it's an inplace operation.
+        }
+        for(const auto &sdr_ptr : inputs) {
+            const auto &data = sdr_ptr->getDense();
+            for(auto z = 0u; z < data.size(); ++z) {
+                dense_[z] = dense_[z] || data[z];
```

Comment:
is it faster this way than adding sdr_ptr->getSparse() to a std::set and then returning that?

Reply:
I operate on the dense format data because I copy pasted from the method `sdr.intersection`.

I did not profile this code, so I can't say with any certainty which algorithms are faster.  If/when we change the SDR to keep its sparse formatted data sorted, then we can also look into optimizing this to work with the sparse data.